swarm/storage/resource.go

```diff
@@ -320,17 +327,19 @@ func (self *ResourceHandler) NewResource(ctx context.Context, name string, frequ
 // It is the callers responsibility to make sure that this chunk exists (if the resource
 // update root data was retrieved externally, it typically doesn't)
 //
-//
-func (self *ResourceHandler) LookupVersionByName(ctx context.Context, name string, period uint32, version uint32, refresh bool) (*resource, error) {
-	return self.LookupVersion(ctx, self.nameHash(name), name, period, version, refresh)
+// If maxPeriod is -1, the default QueryMaxPeriod from ResourceHandlerParams will be used
```

Comment:
not very common API, and i am not convinced a default per resource is justified.
but even if it is, i would just use a struct like {limit: true, maxlookups: 6}, if nil falls back to default.
`{limit: false}` could overwrite default limit and all is transparent

Reply:
I guess we will never agree on what is more intuitive. But I'll make the change. And let's keep the default since we're not sure.